test/sagemaker_tests/pytorch/training/conftest.py

```diff
@@ -475,6 +479,21 @@ def skip_pt20_cuda121_tests(request, ecr_image):
             pytest.skip("PyTorch 2.0 + CUDA12.1 image doesn't support current test")
 
 
+@pytest.fixture(autouse=True)
+def skip_p5_tests(request, ecr_image, instance_type):
+    if "p5." in instance_type:
+        framework, image_framework_version = get_framework_and_version_from_tag(ecr_image)
+        job_type = get_job_type_from_image(ecr_image)
+        if "pytorch" not in framework or "inference" in job_type:
+            pytest.skip("Current image doesn't support P5 EC2 instance.")
```

Comment:
I think it is safe to assume that we will only test pytorch training DLCs on the `sagemaker_tests/pytorch/training` tests.

If someone is sufficiently motivated to test an image that isn't pytorch-training, let's not prevent them from doing what they want to do.

Reply:
We have added P5 CB to all RC pipelines, so tests should get skipped on P5 for other images